pkg/broker/broker.go

```diff
@@ -193,26 +193,13 @@ func (a AnsibleBroker) Provision(instanceUUID uuid.UUID, req *ProvisionRequest,
 	// we're being asked to provision.
 	//
 	// if err is not nil, we will just bubble that up
-	if si, err := a.dao.GetServiceInstance(instanceUUID.String()); err == nil {
-		if si.Id.String() == serviceInstance.Id.String() &&
-			reflect.DeepEqual(si.Parameters, serviceInstance.Parameters) {
 
+	if si, err := a.dao.GetServiceInstance(instanceUUID.String()); err == nil {
+		if serviceInstance.IsEqual(si) {
 			a.log.Debug("already have this instance returning 200")
 			return &ProvisionResponse{}, ErrorAlreadyProvisioned
-		} else if si.Id.String() == serviceInstance.Id.String() &&
-			!reflect.DeepEqual(si.Parameters, serviceInstance.Parameters) {
-
-			// TODO: remove these debug statements at some point
-			a.log.Debug("Existing parameters")
-			for k, v := range map[string]interface{}(*si.Parameters) {
-				a.log.Debug("%s = %s", k, v)
-			}
-			a.log.Debug("Incoming parameters")
-			for k, v := range map[string]interface{}(*serviceInstance.Parameters) {
-				a.log.Debug(fmt.Sprintf("%s = %s", k, v))
-			}
-
-			a.log.Info("we have a duplicate instance with identical parameters, returning 409 conflict")
+		} else if si.Id.String() == serviceInstance.Id.String() {
+			a.log.Info("we have a duplicate instance with parameters that differ, returning 409 conflict")
```

Comment:
That sounds good, makes sense about the reasoning, What is your feeling about changing the comparison of Id to the [uuid.Equals]( https://godoc.org/github.com/pborman/uuid#Equal). 

I think this would prevent issues in the future if the library ever went to a something other than what the string function could compare? 
ex.
``` if uuid.Equals(si.Id, serviceInstance.Id) {```